internal/provider/resource_time_rotating.go

```diff
@@ -0,0 +1,683 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/schemavalidator"
+
+	"github.com/hashicorp/terraform-provider-time/internal/modifiers/timemodifier"
+	"github.com/hashicorp/terraform-provider-time/internal/validators/timevalidator"
+)
+
+var (
+	_ resource.Resource                = (*timeRotatingResource)(nil)
+	_ resource.ResourceWithImportState = (*timeRotatingResource)(nil)
+	_ resource.ResourceWithModifyPlan  = (*timeRotatingResource)(nil)
+)
+
+func NewTimeRotatingResource() resource.Resource {
+	return &timeRotatingResource{}
+}
+
+type timeRotatingResource struct{}
+
+func (t timeRotatingResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_rotating"
+}
+
+func (t timeRotatingResource) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: "Manages a rotating time resource, which keeps a rotating UTC timestamp stored in the Terraform " +
+			"state and proposes resource recreation when the locally sourced current time is beyond the rotation time. " +
+			"This rotation only occurs when Terraform is executed, meaning there will be drift between the rotation " +
+			"timestamp and actual rotation. The new rotation timestamp offset includes this drift. " +
+			"This prevents perpetual differences caused by using the [`timestamp()` function](https://www.terraform.io/docs/configuration/functions/timestamp.html) " +
+			"by only forcing a new value on the set cadence.",
+		Attributes: map[string]tfsdk.Attribute{
+			"day": {
+				Description: "Number day of timestamp.",
+				Type:        types.Int64Type,
+				Computed:    true,
+			},
+			"rotation_days": {
+				Description: "Number of days to add to the base timestamp to configure the rotation timestamp. " +
+					"When the current time has passed the rotation timestamp, the resource will trigger recreation. " +
+					"At least one of the 'rotation_' arguments must be configured.",
+				Type:     types.Int64Type,
+				Optional: true,
+				Validators: []tfsdk.AttributeValidator{
+					schemavalidator.AtLeastOneOf(path.MatchRoot("rotation_hours"),
```

Comment:
Nit: Similar note about potentially moving the rotation AtLeastOneOf attribute validation to the ConfigValidators method for the resource.